### PR TITLE
Updated Entity Enrichment - Generic v2.0, Entity Enrichment - Phishing v2.0, Phishing Investigation - Generic v2.0

### PR DESCRIPTION
**Pull request by idovandijk**
## Status
In Progress - needs documentation

## Related Issues
Fixes: https://github.com/demisto/etc/issues/16881

## Description
- Updated `Entity Enrichment - Generic v2.0` to use v2.1 enrichment subplaybooks - which make use of v2 integrations where available.
- Updated `Entity Enrichment - Phishing v2.0` to be on par with `Entity Enrichment - Generic v2.0` changes (in other words - now using v2.1 enrichment playbooks as well)
- Updated `Phishing Investigation - Generic v2.0` to use `Entity Enrichment - Phishing v2.0`

## Related PRs
**Note - will fail until those are merged:**
Account Enrichment
https://github.com/demisto/content/pull/3485
Email Enrichment
https://github.com/demisto/content/pull/3484
Endpoint Enrichment
https://github.com/demisto/content/pull/3501

## Required version of Demisto
4.1.0

## Does it break backward compatibility?
   - Yes - Phishing will now use v2.0 integrations since it uses `Entity Enrichment - Phishing v2.0`, which now uses v2 integrations where available.

## Must have
- [x] Tests
- [x] Documentation
- [ ] Code Review

### CODE REVIEW

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 1]
```diff
@@ -479,113 +476,110 @@ inputs:
   description: A list of internal domains, separated by commas. The list will be used
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: A CSV list of internal domains.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 2]
```diff
@@ -479,113 +476,110 @@ inputs:
...
     to determine whether an email address is internal or external.
 outputs:
-- contextPath: Account
-  description: The Account's object
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The account object.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 3]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-  type: unknown
-- contextPath: Account.ID
-  description: The unique Account DN (Distinguished Name)
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The unique account DN (Distinguished Name).

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 4]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-  type: string
-- contextPath: Domain
-  description: The domain objects
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The domain objects.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 5]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-  type: unknown
-- contextPath: URL
-  description: The URL's object
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The URL object.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 6]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-  type: unknown
-- contextPath: URL.Malicious
-  description: whether url was detected as malicious
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: Whether a URL was detected as malicious.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 7]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-  type: string
-- contextPath: Account.Email.Address
-  description: The Email account full address
-  type: string
 - contextPath: IP
-  description: The IP objects
+  description: The IP object.
   type: unknown
-- contextPath: Account.Email.Domain
-  description: The Email account domain
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The email account domain.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 8]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-  type: string
-- contextPath: Account.Email.NetworkType
-  description: The Email account NetworkType (could be Internal/External)
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The email account NetworkType (Internal/External).

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 9]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-  type: string
-- contextPath: Account.Email.Username
-  description: The Email account username
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The username associated with the email account.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 10]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-  type: string
-- contextPath: Account.Email.Distance.Domain
-  description: The compared domain
+- contextPath: Endpoint
+  description: The endpoint object.
   type: unknown
-- contextPath: Account.Email.Distance.Value
-  description: 'The distance between the email domain and the compared domain '
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: 'The distance between the email domain and the compared domain. '

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 11]
```diff
@@ -479,113 +476,110 @@ inputs:
...
+- contextPath: Endpoint.Hostname
+  description: The hostname that was enriched.
   type: string
-- contextPath: Account.Type
-  description: Type of the Account entity
+- contextPath: Endpoint.OS
+  description: The endpoint's operating system.
   type: string
-- contextPath: Account.Username
-  description: The Account username
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The account username.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 12]
```diff
@@ -479,113 +476,110 @@ inputs:
...
+- contextPath: Endpoint.IP
+  description: A list of endpoint IP addresses.
+- contextPath: Endpoint.MAC
+  description: A list of endpoint MAC addresses.
+- contextPath: Endpoint.Domain
+  description: The endpoint domain name.
   type: string
-- contextPath: Account.Email
-  description: The email address associated with the Account
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The email address associated with the account.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 13]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-- contextPath: Account.Groups
-  description: The groups the Account is part of
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The groups that the account is a member of.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 14]
```diff
@@ -479,113 +476,110 @@ inputs:
...
+- contextPath: DBotScore
+  description: The DBotScore object.
   type: unknown
-- contextPath: Account.DisplayName
-  description: The Account display name
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The account display name.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 15]
```diff
@@ -479,113 +476,110 @@ inputs:
...
+- contextPath: DBotScore.Indicator
+  description: The indicator that was tested.
+  type: string
+- contextPath: DBotScore.Type
+  description: The indicator type.
   type: string
-- contextPath: Account.Manager
-  description: The Account's manager
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The account's manager.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 16]
```diff
@@ -479,113 +476,110 @@ inputs:
...
+- contextPath: File.MD5
+  description: MD5 hash of the file.
   type: string
-- contextPath: Endpoint
-  description: The Endpoint's object
+- contextPath: File.Malicious
+  description: Whether the file is malicious.
   type: unknown
-- contextPath: Endpoint.Hostname
-  description: The hostname to enrich
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The hostname to enrich.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 17]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-  type: string
-- contextPath: Endpoint.OS
-  description: Endpoint OS
+- contextPath: File.Malicious.Vendor
+  description: For malicious files, the vendor that made the decision.
   type: string
-- contextPath: Endpoint.IP
-  description: List of endpoint IP addresses
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: List of endpoint IP addresses.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 18]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-- contextPath: Endpoint.MAC
-  description: List of endpoint MAC addresses
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: List of endpoint MAC addresses.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 19]
```diff
@@ -479,113 +476,110 @@ inputs:
...
-- contextPath: Endpoint.Domain
-  description: Endpoint domain name
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The endpoint domain name.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml [hunk 20]
```diff
@@ -479,113 +476,110 @@ inputs:
...
+  description: The account object.
+  type: unknown
+- contextPath: Account.Email
+  description: The email of the account.
+  type: unknown
+- contextPath: Account.Email.NetworkType
+  description: The email account NetworkType (Internal/External).
   type: string
-- contextPath: DBotScore.Score
-  description: The DBot score
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The DBot score.

File: Playbooks/playbook-Entity_Enrichment_-_Phishing_v2.yml [hunk 21]
```diff
@@ -394,113 +392,110 @@ inputs:
   description: A list of internal domains, separated by commas. The list will be used
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Phishing_v2.yml]: A CSV list of internal domains.

File: Playbooks/playbook-Entity_Enrichment_-_Phishing_v2.yml [hunk 22]
```diff
@@ -394,113 +392,110 @@ inputs:
...
     to determine whether an email address is internal or external.
 outputs:
-- contextPath: Account
-  description: The Account's object
```
Comment (kirbles19) [inline Playbooks/playbook-Entity_Enrichment_-_Phishing_v2.yml]: For all of these descriptions, see above.

File: Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml (hunk 2 above)
Comment (idovandijk) [reply Playbooks/playbook-Entity_Enrichment_-_Generic_v2.yml]: The above are deleted lines :)

**Pull request merged**